src/fable-library/Uri.ts

```diff
@@ -1,127 +1,208 @@
+import { FSharpRef } from "./Types";
+
 export const enum UriKind {
   RelativeOrAbsolute = 0,
   Absolute = 1,
   Relative = 2,
 }
 
+type State =
+  | { value: URL; kind: UriKind.Absolute }
+  | { value: string; kind: UriKind.Relative | UriKind.RelativeOrAbsolute };
+
+type Result<T> = { tag: "ok"; value: T } | { tag: "error"; error: string };
```

Comment:
I'm wondering whether this construct (or similar) is defined else where in here so that I don't need to define this internal helper type just this usage.

Reply:
The Result type is defined now in F#... in the Choice.fs file :/ So it's ok to have your custom `Result` here internally :+1:

Reply:
Cool, just wanted to check that there wasn't another internal `Result` defined somewhere else for the typescript stuff.

Reply:
Now that I think of it, maybe we could put it in `Types.ts` with a note saying it's only an internal type. But I can do that :+1: